phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -2501,6 +2503,33 @@ public Pair<Integer, ResultSet> executeAtomicUpdateReturnRow(String sql) throws
         return result;
     }
 
+    /**
+     * Executes the given SQL statement similar to JDBC API executeUpdate() but also returns the
+     * old row (before update) as Result object back to the client. This must be used with
+     * auto-commit Connection. This makes the operation atomic.
+     * Return the old row (state before update) regardless of whether the update is
+     * successful or not.
+     *
+     * @param sql The SQL DML statement, UPSERT or DELETE for Phoenix.
+     * @return The pair of int and ResultSet, where int represents value 1 for successful row
+     * update and 0 for non-successful row update, and ResultSet represents the old state of the
+     * row.
+     * @throws SQLException If the statement cannot be executed.
+     */
+    // Note: Do Not remove this, it is expected to be used by downstream applications
+    public Pair<Integer, ResultSet> executeAtomicUpdateReturnOldRow(String sql)
```

Comment:
The only difference between this and the existing `executeAtomicUpdateReturnRow` is the parameter `ReturnResult` correct? Why not refactor to avoid the duplication?

Reply:
Yes, I was thinking about it, but I thought this would be more clear.
Using Phoenix internal Enum for the end user API might be bit odd? WDYT?

Reply:
This is anyway an internal API (i.e., non-JDBC API) that we are exposing, so it is just a matter of committing the enum to be public and stable, so I don't see that as an issue.

Reply:
True, but see my other [comment](https://github.com/apache/phoenix/pull/2199#discussion_r2163102267) which is applicable here too.